hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/LogFileIterator.scala

```diff
@@ -240,17 +236,22 @@ class RecordMergingFileIterator(split: HoodieMergeOnReadFileSplit,
   private def merge(curRow: InternalRow, newRecord: HoodieRecord[_]): Option[InternalRow] = {
     // NOTE: We have to pass in Avro Schema used to read from Delta Log file since we invoke combining API
     //       on the record from the Delta Log
+    val curRecord = recordMerger.getRecordType match {
```

Comment:
I don't think it makes sense to extract this one (we have identical matches right on top of each other)